### ISSUE TITLE
False positive when not all arguments are typed and last argument has default value None

### ISSUE THREAD

**Issue by stardust85** [connor-makowski/type_enforced#19]
Package version: 0.0.15
Python version: 3.10.6

code:
```python
@Enforcer
def my_func(first, second: str = 'second', third=None):
    print(first)
    print(second)
    print(third)

my_func(first='first')
```

Expected output (as you get without the decorator):
```
first
second
None
```

Current output:
```
Traceback (most recent call last):
  File "/home/ms580j/leam/backend/utils/export/type_enforced_bug.py", line 26, in <module>
    my_func(first='first')
  File "/home/ms580j/.local/share/virtualenvs/export-pPyZbvWm/lib/python3.10/site-packages/type_enforced/enforcer.py", line 95, in __call__
    self.__check_type__(assigned_vars.get(key), value, key)
  File "/home/ms580j/.local/share/virtualenvs/export-pPyZbvWm/lib/python3.10/site-packages/type_enforced/enforcer.py", line 117, in __check_type__
    self.__exception__(
  File "/home/ms580j/.local/share/virtualenvs/export-pPyZbvWm/lib/python3.10/site-packages/type_enforced/enforcer.py", line 34, in __exception__
    raise TypeError(f"({self.__fn__.__qualname__}): {message}")
TypeError: (my_func): Type mismatch for typed variable `second`. Expected one of the following `[<class 'str'>]` but got `<class 'NoneType'>` instead.
```

**Comment by connor-makowski** [comment]
Thanks for creating this issue. Let me look into this and see what is going on.

**Comment by connor-makowski** [comment]
Solved with 0.0.16 which is now live. Can you confirm the issue is resolved on your side?

**Comment by stardust85** [comment]
Awesome, thank you for the super-fast fix! Confirming that the issue is fixed in 0.0.16 on my side :)
